packages/contracts-periphery/contracts/universal/op-nft/OptimistInviter.sol

```diff
@@ -0,0 +1,273 @@
+    bytes32 public constant CAN_MINT_FROM_INVITE_ATTESTATION_KEY =
+        keccak256("optimist.can-mint-from-invite");
+
+    /**
+     * @notice Granter who can set accounts' invite counts.
+     */
+    address public immutable INVITE_GRANTER;
+
+    /**
+     * @notice Address of the AttestationStation contract.
+     */
+    AttestationStation public immutable ATTESTATION_STATION;
+
+    /**
+     * @notice Struct that represents a claimable invite that will be signed by the issuer.
+     *
+     * @custom:field issuer   Address that issued the signature. Reason this is explicitly included,
+     *                        and not implicitly assumed to be the recovered address from the
+     *                        signature is that the issuer may be using a ERC-1271 compatible
+     *                        contract wallet, where the recovered address is not the same as the
+     *                        issuer, or the signature is not an ECDSA signature at all.
+     * @custom:field nonce    Pseudorandom nonce to prevent replay attacks.
+     */
+    struct ClaimableInvite {
+        address issuer;
+        bytes32 nonce;
+    }
+
+    /**
+     * @notice Maps from hashes to whether or not they have been committed.
+     */
+    mapping(bytes32 => bool) public commitments;
+
+    /**
+     * @notice Maps from addresses to nonces to whether or not they have been used.
+     */
+    mapping(address => mapping(bytes32 => bool)) public usedNonces;
+
+    /**
+     * @notice Maps from addresses to number of invites they have.
+     */
+    mapping(address => uint256) public inviteCounts;
+
+    /**
+     * @custom:semver 1.0.0
+     *
+     * @param _inviteGranter      Address of the invite granter.
+     * @param _attestationStation Address of the AttestationStation contract.
+     */
+    constructor(address _inviteGranter, AttestationStation _attestationStation) Semver(1, 0, 0) {
+        INVITE_GRANTER = _inviteGranter;
+        ATTESTATION_STATION = _attestationStation;
+    }
+
+    /**
+     * @notice Initializes this contract, setting the EIP712 context.
+     *
+     *         Only update the EIP712_VERSION when there is a change to the signature scheme.
+     *         After the EIP712 version is changed, any signatures issued off-chain but not
+     *         claimed yet will no longer be accepted by the claimInvite function. Please make
+     *         sure to notify the issuers that they must re-issue their invite signatures.
+     *
+     * @param _name Contract name
+     */
+    function initialize(string memory _name) public initializer {
+        __EIP712_init(_name, EIP712_VERSION);
+    }
+
+    /**
+     * @notice Allows invite granter to set the number of invites an address has.
+     *
+     * @param _accounts    An array of accounts to update the invite counts of.
+     * @param _inviteCount Number of invites to set to.
+     */
+    function setInviteCounts(address[] calldata _accounts, uint256 _inviteCount) public {
```

Comment:
Do we generally expect every account to have the same number of invites? If so, this is fine. If not, we could consider making this an array of structs of (account, inviteCount). Just depends on how we expect to use it.

Reply:
Yeah, the main we expect to use it is to issue everyone 3 invites.